source/scripts/main.js

```diff
@@ -0,0 +1,196 @@
+    const recipeCards = document.getElementById("recipe-card-container");
+    recipeCards.style.visibility = "visible";
+}
+function hideRecipeCards() {
+    const recipeCards = document.getElementById("recipe-card-container");
+    recipeCards.style.visibility = "hidden";
+}
+
+function showCategoryCards() {
+    const categoryCards = document.getElementById("category-wrapper");
+    categoryCards.style.visibility = "visible";
+}
+
+function hideCategoryCards() {
+    const categoryCards = document.getElementById("category-wrapper");
+    categoryCards.style.visibility = "hidden";
+}
+
+function updateSettings() {
+    const dietaryRestrictionList = [];
+    // Get all the inputs under the div
+    const dietaryContainerElements = document.getElementById('dietary-container').elements;
+    for (let i = 0; i < dietaryContainerElements.length; i++) {
+        // If a checkbox is checked, then add it to our list
+        const inputElement = dietaryContainerElements[i];
+        if (inputElement.checked) {
+            dietaryRestrictionList.push(inputElement.value);
+        }
+    }
+
+    const intolerancesRestrictionsList = [];
+    const intolerancesContainerElements = document.getElementById("intolerances-container");
+    for (let i = 0; i < intolerancesContainerElements.length; i++) {
+        // If a checkbox is checked, then add it to our list
+        const inputElement = intolerancesContainerElements[i];
+        if (inputElement.checked) {
+            intolerancesRestrictionsList.push(inputElement.value);
+        }
+    }
+
+    // Add lists to local storage
+    localStorage.setItem("dietaryRestrictions", JSON.stringify(dietaryRestrictionList));
+    localStorage.setItem("intolerancesRestrictions", JSON.stringify(intolerancesRestrictionsList));
+
+    // TODO: add confirmation message in HTML (alert is temporary)
+    alert("your preferences have been updated");
+}
+
+/**
+ * This function checks localStorage for saved recipes, and then display bookmark-filled for the ones that already saved
+ * @param {Object} data 
+ * @returns None
+ */
+function checkBookMark(data) {
+    let bookmarkList = JSON.parse(localStorage.getItem("bookmark"));
+    const title = data["title"];
+    if (bookmarkList == null)
+        return;
+    if (bookmarkList[title] != null) {
+        let bookMark = document.querySelector("#recipe-page-container > recipe-page").shadowRoot.querySelector("#bookmark");
+        bookMark.src = "./img/icons/bookmark-filled.svg";
+        bookMark.setAttribute("name", "bookmark-filled");
+    }
+}
+
+/**
+ * When user clicks on bookMark icon, it saves recipe's title and ID to localStorage
+ * If the recipe already saved, clicking it again will remove it from localStorage.
+ * The data being stored in 'bookmark'
+ */
+function setBookMark() {
+    // check local storage for bookmark
+    let bookmarkList = JSON.parse(localStorage.getItem("bookmark"));
+    if (bookmarkList == null)
+        bookmarkList = {};
```

Comment:
Codacy has a fix for the issue: [Expected { after 'if' condition.](https://app.codacy.com/gh/cse110-fa21-group23/cse110-fa21-group23/pullRequest?prid=8422335)
```suggestion
        {bookmarkList = {};}
```